src/ui-components/WalletConnectModal.tsx

```diff
@@ -0,0 +1,448 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import React, { useContext, useEffect, useState } from 'react';
+import { Alert, Button, Form, Modal, Spin } from 'antd';
+import { poppins } from 'pages/_app';
+import { NotificationStatus, Wallet } from '~src/types';
+import { ApiContext } from '~src/context/ApiContext';
+import { useUserDetailsContext } from '~src/context';
+import { NetworkContext } from '~src/context/NetworkContext';
+import WalletButton from '~src/components/WalletButton';
+import { LoadingOutlined } from '@ant-design/icons';
+import { WalletIcon } from '~src/components/Login/MetamaskLogin';
+import AccountSelectionForm from '~src/ui-components/AccountSelectionForm';
+import { isWeb3Injected } from '@polkadot/extension-dapp';
+import { Injected, InjectedAccount, InjectedWindow } from '@polkadot/extension-inject/types';
+import getEncodedAddress from '~src/util/getEncodedAddress';
+import { inputToBn } from '~src/util/inputToBn';
+import BN from 'bn.js';
+import { APPNAME } from '~src/global/appName';
+import styled from 'styled-components';
+import getSubstrateAddress from '~src/util/getSubstrateAddress';
+import { InjectedTypeWithCouncilBoolean } from './AddressDropdown';
+import { EAddressOtherTextType } from './Address';
+import ConnectAddressIcon from '~assets/icons/connect-address.svg';
+import CloseIcon from '~assets/icons/close.svg';
+import nextApiClientFetch from '~src/util/nextApiClientFetch';
+import queueNotification from './QueueNotification';
+import cleanError from '~src/util/cleanError';
+import { ChallengeMessage, ChangeResponseType } from '~src/auth/types';
+import { handleTokenChange } from '~src/services/auth.service';
+import { stringToHex } from '@polkadot/util';
+
+interface Props{
+  className?: string;
+  open: boolean;
+  setOpen: (pre: boolean) => void;
+  closable?: boolean;
+  walletKey: string;
+  addressKey: string;
+  onConfirm?: () => void;
+  LinkAddressNeeded?: boolean;
+}
+
+const ZERO_BN = new BN(0);
+
+const WalletConnectModal = ({ className, open, setOpen, closable, walletKey, addressKey, onConfirm, LinkAddressNeeded }: Props) => {
```

Comment:
WalletConnectModal -> ConnectWalletModal

Clashes with the walletConnect wallet